app/src/main/java/com/portfolio/tasky/entry/EntryActivity.kt

```diff
@@ -2,28 +2,75 @@ package com.portfolio.tasky.entry
 import androidx.appcompat.app.AppCompatActivity
-import androidx.appcompat.widget.Toolbar
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
 import androidx.databinding.DataBindingUtil
 import com.portfolio.tasky.*
 import com.portfolio.tasky.databinding.ActivityEntryBinding
 import com.portfolio.tasky.entry.fragments.LoginFragment
-import com.portfolio.tasky.usecases.FragmentInflater
-import com.portfolio.tasky.usecases.FragmentInflaterImpl
-import com.portfolio.tasky.usecases.ToolbarHandler
-import com.portfolio.tasky.usecases.ToolbarHandlerImpl
+import com.portfolio.tasky.globals.Constants
+import com.portfolio.tasky.usecases.*
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class EntryActivity : AppCompatActivity(), FragmentInflater by FragmentInflaterImpl(), ToolbarHandler by ToolbarHandlerImpl() {
-    private var viewBinding: ActivityEntryBinding? = null
+    private lateinit var viewBinding: ActivityEntryBinding
+    private lateinit var networkChangeReceiver : NetworkChangeReceiver
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         viewBinding = DataBindingUtil.setContentView(this, R.layout.activity_entry)
 
+        setObservers()
         startLoginFragment()
     }
 
+    private fun setObservers() {
+        networkChangeReceiver = NetworkChangeReceiver(this)
+        networkChangeReceiver.observe(this){
+            when(it){
+                NetworkStatus.Available -> networkAvailability(true)
+                NetworkStatus.Unavailable -> networkAvailability(false)
+            }
+        }
+    }
+
+    private fun networkAvailability(isAvailable : Boolean){
+        onConnected(viewBinding.layoutOnlineMode.tvOnlineMode, isAvailable)
+        if(isAvailable){
+            viewBinding.layoutOnlineMode.drawableFirst = ContextCompat.getDrawable(this, R.drawable.ic_online)
+        }else{
+            viewBinding.layoutOnlineMode.drawableFirst = ContextCompat.getDrawable(this, R.drawable.ic_offline)
+        }
+    }
+
+     private fun onConnected(offlineTextField: AppCompatTextView, connected: Boolean) {
+        if (connected){
+            offlineTextField.text = Constants.Companion.AppConnectivityMode.ONLINE
+        }else{
+            offlineTextField.text = Constants.Companion.AppConnectivityMode.OFFLINE
+        }
+        animate(offlineTextField, connected)
+    }
+
+    private fun animate(offlineTextField: AppCompatTextView, connected: Boolean) {
+        offlineTextField.visibility = View.VISIBLE
+        val endToStartAnimation = TranslateAnimation(500f, 0f, 0f, 0f)
+        endToStartAnimation.duration = Constants.Companion.AnimationProperties.DURATION
+        offlineTextField.startAnimation(endToStartAnimation)
+
+        Handler(Looper.getMainLooper()).postDelayed({
+            offlineTextField.text = ""
+            if (connected){
+                offlineTextField.visibility =  View.GONE
```

Comment:
Instead of this, you can also just do `offlineTextField.isVisible = false` (false for GONE, true for VISIBLE). It's usually easier to work with that since you can directly assign a boolean to it.